src/test/kotlin/com/exactpro/th2/codec/ProcessorTest.kt

```diff
@@ -110,6 +110,60 @@ class ProcessorTest {
         Assertions.assertEquals(2, result.groupsCount) {"Wrong batch size"}
     }
 
+    @Test
+    fun `load test - decode`() {
+        val processor = DecodeProcessor(TestCodec(false), ORIGINAL_PROTOCOLS) { _, _ -> }
+
+        var result: MessageGroupBatch? = null
+        for (i in 0..3000 step 10) {
+            result = processor.process(makeRawMessageBatch(i.toLong()))
+        }
+
+        Assertions.assertEquals(10, result?.groupsCount) {"Wrong batch size"}
```

Comment:
This should be inside of `for` loop - otherwise we would be checking only the last result.
We should also check that group order was not modified.